Source/Particles/ParticleCreation/FilterCreateTransformFromFAB.H

```diff
@@ -0,0 +1,206 @@
+                create2(dst2_data, N*p_offsets[i] + dst2_index + n, x, y, z);
+            }
+            transform(dst1_data, dst2_data, N*p_offsets[i] + dst1_index,
+                    N*p_offsets[i] + dst2_index, N, arrNumPartCreation(j,k,l));
+        }
+    });
+
+    Gpu::streamSynchronize();
+    return num_added;
+}
+
+/**
+ * \brief Apply a filter on a list of FABs, then create and apply a transform
+ * operation to the particles depending on the output of the filter.
+ *
+ * This version of the function takes as input a filter functor (and an array of FABs that
+ * can be used in the filter functor), uses it to obtain a mask and a FAB and then calls another
+ * version of filterCreateTransformFromFAB that takes the mask and the FAB as inputs.
+ *
+ * \tparam N number of particles created in the dst(s) in each cell
+ * \tparam DstTile the dst particle tile type
+ * \tparam FABs the src array of Array4 type
+ * \tparam Index the index type, e.g. unsigned int
+ * \tparam FilterFunc the filter function type
+ * \tparam CreateFunc1 the create function type for dst1
+ * \tparam CreateFunc2 the create function type for dst2
+ * \tparam TransFunc the transform function type
+ *
+ * \param[in,out] dst1 the first destination tile
+ * \param[in,out] dst2 the second destination tile
+ * \param[in] box the box where the particles are created
+ * \param[in] src_FABs An Array of Array4 (e.g. EM fields) defined on box on which
+ *            the filter operation is applied/
+ * \param[in] dst1_index the location at which to starting writing the result to dst1
+ * \param[in] dst2_index the location at which to starting writing the result to dst2
+ * \param[in] filter a callable returning a value > 0 if particles are to be created
+ *            in the considered cell.
+ * \param[in] create1 callable that defines what will be done for the create step for dst1.
+ * \param[in] create2 callable that defines what will be done for the create step for dst2.
+ * \param[in] transform callable that defines the transformation to apply on dst1 and dst2.
+ *
+ * \return num_added the number of particles that were written to dst1 and dst2.
+ */
+template <int N, typename DstTile, typename FABs, typename Index,
+          typename FilterFunc, typename CreateFunc1, typename CreateFunc2,
+           typename TransFunc>
+Index filterCreateTransformFromFAB (DstTile& dst1, DstTile& dst2, const amrex::Box box,
+                                const FABs& src_FABs, const Index dst1_index,
+                                const Index dst2_index, FilterFunc&& filter,
+                                CreateFunc1&& create1, CreateFunc2&& create2,
+                                TransFunc && transform) noexcept
+{
+    using namespace amrex;
+
+    FArrayBox NumPartCreation(box, 1);
+    Elixir tmp_eli = NumPartCreation.elixir(); // Do we actually need this elixir?
+                            // The tests on GPU seem to be working well without it.
```

Comment:
It is probably not required because you added an explicit `synchronize` in the other `filterCreateTransformFromFAB`. I would be in favor in keeping it though, with a comment
```suggestion
    // This may be unnecessary because of the Gpu::streamSynchronize() in
    // filterCreateTransformFromFAB called below, but let us keep it for safety.
    Elixir tmp_eli = NumPartCreation.elixir();
```